Pod/Classes/OpaqueImageViewComponent/AROpaqueImageView.m

```diff
@@ -74,36 +75,41 @@ - (void)setImageURL:(NSURL *)imageURL;
     return;
   }
 
-  // TODO Setting decompress to NO, because Eigen sets it to YES.
-  //      We need to send a PR to SDWebImage to disable decoding
-  //      with an option to the download method.
+  // Once an image has been cached, it will be returned immediately, which means it might try to resize the image on a
+  // view with zero size. Give the (React) layout code a chance to assign a frame first.
   //
-  SDWebImageManager *manager = [SDWebImageManager sharedManager];
-  manager.imageCache.shouldDecompressImages = NO;
-  manager.imageDownloader.shouldDecompressImages = NO;
-
-  __weak typeof(self) weakSelf = self;
-  self.downloadOperation = [manager downloadImageWithURL:self.imageURL
-                                                  options:0
-                                                progress:nil
-                                                completed:^(UIImage *image,
-                                                            NSError *_,
-                                                            SDImageCacheType __,
-                                                            BOOL ____,
-                                                            NSURL *imageURL) {
-    __strong typeof(weakSelf) strongSelf = weakSelf;
-    // Only really assign if the URL we downloaded still matches `self.imageURL`.
-    if (strongSelf && [imageURL isEqual:strongSelf.imageURL]) {
-      // The view might not yet be associated with a window, in which case
-      // the view will always return 1 for contentScaleFactor.
-      CGFloat scaleFactor = [[UIScreen mainScreen] scale];
-      LoadImage(image, strongSelf.bounds.size, scaleFactor, ^(UIImage *loadedImage) {
-        if ([imageURL isEqual:weakSelf.imageURL]) {
-          weakSelf.image = loadedImage;
-        }
-      });
-    }
-  }];
+  dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
> React won't release the main queue

I’m not sure what you mean by this.

But what happened here was that the image was loaded (and we tried to do the above loading into memory and resizing) _before_ RN had done its layout pass, which meant dimensions were still `0`. This did not occur on the initial load, because at that time the image still needed download/loading into memory, but upon next loads that data was actually already loaded.